src/homework6/modelfromlife.py

```diff
@@ -0,0 +1,275 @@
+        cards = self.blackjack_deck[:] * 4
+        random.shuffle(cards)
+        players = []
+        croupier_hand = [cards.pop(0), cards.pop(1)]
+        for person in persons_list:
+            if person.do_i_have_cash():
+                bet = person.bet_blackjack()
+                hand = [cards.pop(0), cards.pop(1)]
+                one_more = person.get_hand(hand)
+                players.append([person, bet, hand, one_more])
+        while any(param[-1] for param in players):
+            for person in players:
+                if person[-1]:
+                    person[2].append(cards.pop(0))
+                    person[-1] = person[0].get_hand(person[2])
+        while sum(croupier_hand) < 17:
+            croupier_hand.append(cards.pop(0))
+        for person in players:
+            if 22 > sum(person[2]) > sum(croupier_hand):
+                person[0].won(person[1] * 2)
+                self.capital -= person[1]
+            else:
+                self.capital += person[1]
+
+    def bar(self):
+        drinks = {'mojito': (1, 10),
+                  'bud': (1, 5),
+                  'vodka': (2, 10),
+                  'jimbeam': (2, 15),
+                  'chivas': (2, 50)}
+        for person in self.bar_visitors:
+            the_picked_drink = random.choice(person.prefer_drink)
+            print('{} is drinking a glass of {}'
+                  .format(person.name, the_picked_drink))
+            person.drunk += drinks[the_picked_drink][0]
+            person.money -= drinks[the_picked_drink][1]
+            self.capital += drinks[the_picked_drink][1]
+        self.bar_visitors.clear()
+
+    def security(self, person):
+        if person in self.roulette_players:
+            self.roulette_players.remove(person)
+        if person in self.blackjack_players:
+            self.blackjack_players.remove(person)
+        if person in self.bar_visitors:
+            self.bar_visitors.remove(person)
+        self.list_guests.remove(person)
+
+    def __str__(self):
+        return ('Casino profit is {} dollars'
+                .format(self.capital - self.start_capital))
+
+
+class UsualGuest(object):
+
+    def __init__(self, name):
+        self.name = name
+        list_guests.append(self)
+        self.money = 2500
+        self.drunk = 1
+        self.start_bet = 5
+        self.prefer_place = ['roulette', 'blackjack', 'one-hand bandit']
+        self.current_place = ''
+        self.go_to_the_bar = True
+        self.prefer_drink = ['bud', 'jimbeam']
+
+    def choice_place(self):
+        if self.current_place == 'bar' and self.go_to_the_bar:
+            self.go_to_the_bar = False
+        else:
+            self.go_to_the_bar = True
+            self.current_place = random.choice(self.prefer_place)
+
+    def do_i_have_cash(self):
+        if self.money - (self.start_bet * self.drunk) <= 0:
+            self.gameover()
+            return False
+        else:
+            return True
+
+    def won(self, pot):
```

Comment:
it should be moved to games (or to casino class)

Reply:
moved to casino class